src/xunit.performance.api/XunitPerformanceHarness.cs

```diff
@@ -94,27 +75,164 @@ private void WriteStatisticsToFile(CSVMetricReader reader)
                 }
 
                 foreach (var measurement in measurements)
-                {
-                    var metric = measurement.Key;
-                    var count = measurement.Value.Count;
-                    var avg = measurement.Value.Average();
-                    var stdev_s = Math.Sqrt(measurement.Value.Sum(x => Math.Pow(x - avg, 2)) / (measurement.Value.Count - 1));
-                    var max = measurement.Value.Max();
-                    var min = measurement.Value.Min();
-
-                    var r = statisticsTable.AppendRow();
-                    r[col0_testName] = testCaseName;
-                    r[col1_metric] = metric;
-                    r[col2_iterations] = count.ToString();
-                    r[col3_average] = avg.ToString();
-                    r[col4_stdevs] = stdev_s.ToString();
-                    r[col5_min] = min.ToString();
-                    r[col6_max] = max.ToString();
-                }
+                    yield return (testCaseName, measurement.Key, measurement.Value);
             }
+        }
+
+        private static DataTable CreateStatistics(CSVMetricReader reader)
```

Comment:
Yes, I agree. The whole file output story of the API needs some cleaning, and this will not be here moving forward. This was just to unblock corefxlabs and provide their feature requests.

---
In reply to: [100353294](https://github.com/Microsoft/xunit-performance/pull/183#discussion_r100353294) [](ancestors = 100353294)